artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/proton/AMQPConnectionContext.java

```diff
@@ -324,10 +344,11 @@ protected void remoteLinkOpened(Link link) throws Exception {
          } else {
             if (isReplicaTarget(receiver)) {
                try {
-                  protonSession.getSessionSPI().check(SimpleString.toSimpleString(ProtonProtocolManager.MIRROR_ADDRESS), CheckType.SEND, getSecurityAuth());
+                  protonSession.getSessionSPI().check(SimpleString.toSimpleString(link.getTarget().getAddress()), CheckType.SEND, getSecurityAuth());
                } catch (ActiveMQSecurityException e) {
                   throw ActiveMQAMQPProtocolMessageBundle.BUNDLE.securityErrorCreatingProducer(e.getMessage());
                }
+               receiver.setOfferedCapabilities(new Symbol[]{AMQPMirrorControllerSource.MIRROR_CAPABILITY});
```

Comment:
It should verify it was a desired capability and refuse the links that dont have it set, i.e old brokers that dont know about it, or future brokers should they change in incompatible ways and update the capability. (or current brokers whos info got stripped by a router intermediary for now)

Reply:
At this point we should just reject old brokers. The protocol is not really compatible. Always rejecting would be the right thing to be done here.

Reply:
Yep, that was a key part of the suggestion for adding capabilities...plus also for future use if it is made incompatible again in some way.

Just saying that if the other side didnt 'desire' the capability then you shoudlnt really offer it to them, they arent allowed to use it at that point regardless (it has to be desired by, and offered to, a peer to indicate they can use it).

Reply:
I understood now.. I see.. thanks